src/placement/jump_map.c

```diff
@@ -1061,11 +1063,96 @@ jump_map_obj_find_rebuild(struct pl_map *map, struct daos_obj_md *md,
 
 static int
 jump_map_obj_find_reint(struct pl_map *map, struct daos_obj_md *md,
-			struct daos_obj_shard_md *shard_md,
-			struct pl_target_grp *tgp_reint, uint32_t *tgt_reint)
+			 struct daos_obj_shard_md *shard_md,
+			 uint32_t reint_ver, uint32_t *tgt_id,
+			 uint32_t *shard_idx, unsigned int array_size,
+			 int myrank)
 {
-	D_ERROR("Unsupported\n");
-	return -DER_NOSYS;
+	struct pl_jump_map           	*jmap;
+	struct pl_obj_layout            *layout;
+	struct pl_obj_layout            *reint_layout;
+	d_list_t                        remap_list;
+	d_list_t                        reint_list;
+	struct jm_obj_placement		jop;
+	int				index;
+	int                             rc;
+
+	int idx = 0;
+
+	D_DEBUG(DB_PL, "Finding Rebuild\n");
+
+	/* Caller should guarantee the pl_map is up-to-date */
+	if (pl_map_version(map) < reint_ver) {
+		D_ERROR("pl_map version(%u) < rebuild version(%u)\n",
+			pl_map_version(map), reint_ver);
+		return -DER_INVAL;
+	}
+
+	jmap = pl_map2mplmap(map);
+
+	rc = jm_obj_placement_get(jmap, md, shard_md, &jop);
+	if (rc) {
+		D_ERROR("jm_obj_placement_get failed, rc %d.\n", rc);
+		return rc;
+	}
+
+	if (jop.jmop_grp_size == 1) {
+		D_DEBUG(DB_PL, "Not replicated object "DF_OID"\n",
+			DP_OID(md->omd_id));
+		return 0;
+	}
+
+	/* Allocate space to hold the layout */
+	rc = pl_obj_layout_alloc(jop.jmop_grp_size * jop.jmop_grp_nr, &layout);
+	if (rc)
+		return rc;
+
+	rc = pl_obj_layout_alloc(jop.jmop_grp_size * jop.jmop_grp_nr, &reint_layout);
+	if (rc)
+		return rc;
+	reint_layout->ol_grp_nr = jop.jmop_grp_nr;
+	reint_layout->ol_grp_size = jop.jmop_grp_size;
+	layout->ol_grp_nr = jop.jmop_grp_nr;
+	layout->ol_grp_size = jop.jmop_grp_size;
+
+
+	D_INIT_LIST_HEAD(&remap_list);
+	D_INIT_LIST_HEAD(&reint_list);
+
+	/* Find targets affected directly by reintegration. */
+	rc = get_object_layout(jmap, layout, &jop, &remap_list, false, md);
+	if (rc)
+		goto out;
+
+	 /* Clear list for next placement operation. */
+        remap_list_free_all(&remap_list);
+        D_INIT_LIST_HEAD(&remap_list);
+
+	/* Find targets indirectly affected by reintegration. */
+	rc = get_object_layout(jmap, reint_layout, &jop, &remap_list, true, md);
+	if (rc)
+		goto out;
+
+	for(index = 0; index < layout->ol_nr; ++index) {
+                uint32_t reint_target = reint_layout->ol_shards[index].po_target;
+                uint32_t original_target = layout->ol_shards[index].po_target;
+                struct pool_target *temp_tgt;
+
+                if(reint_target != original_target) {
```

Comment:
(style)  code indent should use tabs where possible
(style)  space required before the open parenthesis '('